flang/include/flang/Lower/PFTBuilder.h

```diff
@@ -455,11 +466,20 @@ struct FunctionLikeUnit : public ProgramUnit {
   LabelEvalMap labelEvaluationMap;
   SymbolLabelMap assignSymbolLabelMap;
   std::list<FunctionLikeUnit> nestedFunctions;
-  /// Symbol associated to this FunctionLikeUnit.
-  /// Null if the FunctionLikeUnit is an anonymous program.
-  /// The symbol has MainProgramDetails for named programs, otherwise it has
-  /// SubprogramDetails.
-  const semantics::Symbol *symbol{nullptr};
+  /// <Symbol, Evaluation> pairs for each entry point.  The pair at index 0
+  /// is the primary entry point; remaining pairs are alternate entry points.
+  /// The primary entry point symbol is Null for an anonymous program.
+  /// A named program symbol has MainProgramDetails.  Other symbols have
+  /// SubprogramDetails.  Evaluations are filled in for alternate entries.
+  llvm::SmallVector<std::pair<const semantics::Symbol *, Evaluation *>, 1>
+      entryPointList{std::pair{nullptr, nullptr}};
```

Comment:
I should add that adding a layer of indirection here might get a little messy, since the main entry point, which is always present, is now largely treated uniformly like the alternates.  That would probably need to change if it was given special status.